language/move-lang/src/parser/syntax.rs

```diff
@@ -1058,31 +1033,18 @@ fn parse_function_decl<'input>(
     // <FunctionDefName>
     let (n, has_params) = parse_name_maybe_args(tokens)?;
     let name = FunctionName(n);
-    let mut type_parameters = vec![];
-    if has_params {
-        while tokens.peek() != Tok::Greater {
-            type_parameters.push(parse_type_parameter(tokens)?);
-            if tokens.peek() == Tok::Greater {
-                break;
-            }
-            consume_token(tokens, Tok::Comma)?;
-        }
-        tokens.advance()?; // consume the ">"
-    }
+    let type_parameters = if has_params {
+        let list = parse_comma_list(tokens, Tok::Greater, parse_type_parameter, true)?;
+        consume_token(tokens, Tok::Greater)?;
+        list
+    } else {
+        vec![]
+    };
 
-    // "(" <Parameter>* ")"
+    // "(" Comma<Parameter> ")"
     consume_token(tokens, Tok::LParen)?;
-    let mut parameters = vec![];
-    while tokens.peek() != Tok::RParen {
-        let v = parse_var(tokens)?;
-        consume_token(tokens, Tok::Colon)?;
-        let t = parse_single_type(tokens)?;
-        if tokens.peek() == Tok::Comma {
-            tokens.advance()?;
-        }
-        parameters.push((v, t));
-    }
-    tokens.advance()?; // consume the ")"
+    let parameters = parse_comma_list(tokens, Tok::RParen, parse_parameter, true)?;
```

Comment:
This one should probably be consistent with call (which it is now, just highlighting for any changes)